#!/bin/bash
#we are using featureCounts to count the reads mapped to the genes in the BAM files generated by STAR alignment we are making a combined matrix so eventually we can use that and perform differential expression analysis
# define paths
GTF_FILE="path_to your_gtf/gencode.v47.annotation.gtf"  # annotation file used here is gencode v47
OUT_DIR="path_for_output/featureCounts_results"  # Output directory where u want to save the results
mkdir -p $OUT_DIR  # Create output directory if it doesn't exist

# Define stage folders
declare -A STAGE_PATHS
STAGE_PATHS["Non-MS_control"]="base_path/Non-MS_control/star_results_nonms"
STAGE_PATHS["PPMS"]="base_path/PPMS/star_results_ppms"
STAGE_PATHS["RRMS"]="base_path/RRMS/star_results_rrms"
STAGE_PATHS["SPMS"]="base_path/SPMS/star_results_spms"

# Run featureCounts for each stage
for STAGE in "${!STAGE_PATHS[@]}"; do
    BAM_DIR="${STAGE_PATHS[$STAGE]}"
    OUT_FILE="$OUT_DIR/${STAGE}_featureCounts.txt"

    echo "Processing stage: $STAGE"

    # Collect all BAM files in an array
    BAM_FILES=($BAM_DIR/*.bam)

    # Check if there are BAM files before running featureCounts
    if [ ${#BAM_FILES[@]} -eq 0 ]; then
        echo "No BAM files found for $STAGE. Skipping..."
        continue
    fi

    # Run featureCounts on BAM files only
    featureCounts -T 8 -p -t exon -g gene_id \
        -a $GTF_FILE \
        -o $OUT_FILE \
        "${BAM_FILES[@]}"

    echo "FeatureCounts for $STAGE saved at $OUT_FILE"
done

echo "All stages processed successfully!"

