# lncRNA_MS_Analysis
Differential analysis of long non-coding rnas from brain tissue in stagewise multiple sclerosis RNA-Seq Dataset

**Transcriptomic Analysis of lncRNAs in Multiple Sclerosis Brain Tissue**

This repository contains a complete reproducible RNA-Seq pipeline for identifying differentially expressed known and novel long non-coding RNAs (lncRNAs) across different stages of Multiple Sclerosis (MS) using human brain tissue samples.

 Objectives of my study:
- Identify differentially expressed lncRNAs across MS subtypes
- Detect stage-specific and shared lncRNA signatures
- Discover novel lncRNAs via transcript assembly
- Generate high-quality visualizations such as volcano plots for differntial analysis


| Step | Task | Script |
|------|------|--------|
| 1Ô∏è‚É£   | FASTQ Extraction & Integrity Check | `fastq_dump_then_md5.sh` |
| 2Ô∏è‚É£   | STAR Alignment & Transcript Assembly | `string_star.sh` |
| 3Ô∏è‚É£   | Read Quantification (FeatureCounts) | `generate_featurecounts.sh` |
| 4Ô∏è‚É£   | Transcript Merging & Novel lncRNA Discovery and extraction | `run_gffcompare_extract_lncRNA.sh` |
| 5Ô∏è‚É£   | Differential Expression & Visualization | `DESeq2_sumtechreps_withvisualisation.R` |


---

## üìå Important Notes

Important Notes Here.
These are the considerations that were taken in my data edit according options and code that fits your data.
NOT ONE SIZE FITS ALL. IS IT SO?


- RNA-Seq data is **reverse-stranded**, confirmed using STAR `--quantMode` output
- STAR alignment was run in **two-pass mode**
- StringTie used for both **assembly and merging**
- FeatureCounts executed with **`-s 2`** to handle stranded data
- Both **known and novel lncRNAs** are quantified
- **Technical replicates** are collapsed after **batch effect checks**
- Metadata and replicate mapping are **generated programmatically**

---

## üìä Visual Outputs

All results are stored in the `results/` folder and include:

- **PCA plots** ‚Äì sample clustering by condition
- **Volcano plots** ‚Äì for each MS subtype comparison
- **MA plots** ‚Äì log2 fold change vs mean expression
- **Heatmaps** ‚Äì of significant lncRNAs
- **Venn diagrams** ‚Äì shared vs unique lncRNAs


---

## üõ† Tools Used

| Tool | Purpose |
|------|---------|
| `fastq-dump`, `FastQC` | Download and quality control |
| `STAR` | Splice-aware alignment |
| `StringTie` | Transcript assembly |
| `featureCounts` | Read quantification |
| `gffcompare` | Novel transcript classification |
| `DESeq2` | Differential analysis |
| `R`, `ggplot2`, `pheatmap`, `EnhancedVolcano` | Plotting & statistical analysis |

---



<img width="1949" height="923" alt="image" src="https://github.com/user-attachments/assets/346b3c70-7589-4fd7-83d2-62ca1e1ab9d0" />

 üßæ Citation

To be added after peer-reviewed publication.  
Repository currently under pre-publication status.  



## üôã‚Äç‚ôÄÔ∏è Contact

For research collaborations or questions, please contact:  
üìß nandinipatel545@gmail.con
